pkg/utils/replicaresource/volcano_provider.go

```diff
@@ -0,0 +1,123 @@
+	pgName := leaderPod.Annotations[volcanov1beta1.KubeGroupNameAnnotationKey]
+	log := ctrl.LoggerFrom(ctx).WithValues("podGroup", pgName, "namespace", lws.Namespace)
+
+	if err := v.client.Get(ctx, types.NamespacedName{Name: pgName, Namespace: lws.Namespace}, &pg); err != nil {
+		if client.IgnoreNotFound(err) != nil {
+			return err
+		}
+		minResources := v.getMinResources(lws)
+		pg = volcanov1beta1.PodGroup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pgName,
+				Namespace: lws.Namespace,
+			},
+			Spec: volcanov1beta1.PodGroupSpec{
+				// Default startupPolicy is LeaderCreated, Leader and Workers Pods are scheduled together, so MinAvailable is set to size
+				MinMember:    *lws.Spec.LeaderWorkerTemplate.Size,
+				MinResources: &minResources,
+			},
+		}
+
+		// If the StartUpPolicy of lws is LeaderReady, set MinMember to 1 to allow the Leader Pod to be scheduled.
+		// However, minResources should still be set to the minResources of the PodGroup (1 Leader + (size-1) Workers).
+		// If the cluster resources are insufficient, scheduling the Leader Pod alone would be meaningless,
+		// as at least one Worker will definitely not be able to be scheduled.
+		if lws.Spec.StartupPolicy == leaderworkerset.LeaderReadyStartupPolicy {
+			pg.Spec.MinMember = 1
+		}
+
+		if queueName, ok := lws.Annotations[volcanov1beta1.QueueNameAnnotationKey]; ok {
+			pg.Spec.Queue = queueName
+		}
+
+		err = ctrl.SetControllerReference(leaderPod, &pg, v.client.Scheme())
+		if err != nil {
+			return err
+		}
+
+		if err = v.client.Create(ctx, &pg); err != nil {
+			return err
+		}
+		log.V(2).Info("Created PodGroup for LeaderWorkerSet")
+	}
+
+	return nil
+}
+
+func (v *VolcanoProvider) InjectPodGroupMetadata(pod *corev1.Pod) error {
+	lwsName := pod.Labels[leaderworkerset.SetNameLabelKey]
+	groupIndex := pod.Labels[leaderworkerset.GroupIndexLabelKey]
+	pod.Annotations[volcanov1beta1.KubeGroupNameAnnotationKey] = GetPodGroupName(lwsName, groupIndex, pod.Labels[leaderworkerset.RevisionKey])
+
+	return nil
+}
+
+// getMinResources calculates the minimum resources needed for the entire PodGroup [1 Leader + (size-1) Worker pods]
+func (v *VolcanoProvider) getMinResources(lws *leaderworkerset.LeaderWorkerSet) corev1.ResourceList {
+	// Calculate leader resources.
+	leaderTemplate := lws.Spec.LeaderWorkerTemplate.LeaderTemplate
+	if leaderTemplate == nil {
+		// If no leader template is specified, use worker template.
+		leaderTemplate = &lws.Spec.LeaderWorkerTemplate.WorkerTemplate
+	}
+	totalResources := resource.PodRequests(&corev1.Pod{Spec: leaderTemplate.Spec}, resource.PodResourcesOptions{})
+
+	// Calculate and add worker resources for (size-1) workers.
+	workerCount := ptr.Deref(lws.Spec.LeaderWorkerTemplate.Size, 1) - 1
+	if workerCount > 0 {
+		workerResources := resource.PodRequests(&corev1.Pod{Spec: lws.Spec.LeaderWorkerTemplate.WorkerTemplate.Spec}, resource.PodResourcesOptions{})
+		for i := int32(0); i < workerCount; i++ {
```

Comment:
Can we use workerResources * (size-1) here?

Reply:
Sorry I don't understand, you mean "minResources = `workerResources` * (size-1)" ? But we don't consider the resource requirement of the leader pod? I thought the resource requirement of leader pod is important